relay-threading/src/metrics.rs

```diff
@@ -53,13 +105,38 @@ impl AsyncPoolMetrics<'_> {
     }
 
     /// Returns the utilization metric for the pool.
+    ///
+    /// The utilization is measured as the amount of busy work performed by each thread when polling
+    /// the futures.
+    ///
+    /// A utilization of 100% indicates that the pool has been doing CPU-bound work for the duration
+    /// of the measurement.
+    /// A utilization of 0% indicates that the pool didn't do any CPU-bound work for the duration
+    /// of the measurement.
+    ///
+    /// Note that this metric is collected and updated for each thread when the main future is polled,
+    /// thus if no work is being done, it will not be updated.
     pub fn utilization(&self) -> f32 {
-        let total_polled_futures: u64 = self
+        let total_utilization: u64 = self
             .threads_metrics
             .iter()
-            .map(|m| m.active_tasks.load(Ordering::Relaxed))
+            .map(|m| m.utilization() as u64)
             .sum();
 
+        (total_utilization as f32 / self.threads_metrics.len() as f32).clamp(0.0, 1.0)
```

Comment:
I think the the outside of the pool we just want to expose a single utilization, which is the maximum of "activity" and utilization here.

If there are no more tasks available or if all of the threads are busy, in both cases it's 100% utilization. For internal metrics (statsd) this distinction is fine.